### ISSUE TITLE
Android SDL Apps not found on IVI

### ISSUE THREAD

**Issue by ashwink11** [smartdevicelink/sdl_java_suite#1701]
### Bug Report
Android SDL Apps not found on IVI

#### Frequency
The issue is not always reproducible, but it was reported multiple times by users. 

#### Steps to recover
The issue recovers if we launch the app in the foreground and reconnect Bluetooth.
I believe, sometimes the App misses Bluetooth intents in mobile devices and hence the issue is seen.

##### Reproduction Steps
In both methods below, the App did not receive Bluetooth connection intents and does not register on IVI.

Method 1: 

1. Connect Bluetooth to IVI 
2. Install HelloSDL App on a mobile device.
3. Launch app on a mobile device
4. Observation: App not registered on IVI since Bluetooth connection intents not received by App.

Method 2:

1. Launch HelloSDL App on a mobile device.
2. Force Stop App on the device.
3. Connect Bluetooth to IVI
4. Relaunch HelloSDL App on a mobile device
4. Observation: Bluetooth connection events not received by App. App not registered on IVI.

##### Expected Behavior
The app should be registered on IVI

##### Observed Behavior
The app does not register and needs to be relaunched on a mobile device and reconnect Bluetooth. 

##### OS & Version Information
* Android Version: Android S8 Android OS 9.0
* SDL Android Version: SDL Library 5.1.1 and latest from GitHub Develop branch
* Testing Against SYNC3
* Test App: HelloSDL App build variant multi_sec_offDebug

**Comment by ashwink11** [comment]
_Potential fix:_ SDL library could retrieve connected Device Mac address in case it misses Bluetooth connection Intents. Please find attached patch file. The patch has an implementation to retrieve the BT Mac address if it is not available. 

[potential_fix.patch.txt](https://github.com/smartdevicelink/sdl_java_suite/files/6559274/potential_fix.patch.txt)

When the mobile device had another app with an old library and connecteddevice in SDLDeviceListener is NULL (condition when app misses BT connection intents). None of the SDL apps were registered on IVI since no app started the SDL router service. The fix attached mitigates this issue. 

```
2021-05-28 09:49:18.173 23842-23842/com.sdl.hellosdlandroid I/Sdl Broadcast Receiver: 5.1.1: Check for local router
2021-05-28 09:49:20.353 23842-23842/com.sdl.hellosdlandroid I/Sdl Broadcast Receiver: 5.1.1: : This app's package: com.sdl.hellosdlandroid
2021-05-28 09:49:20.354 23842-23842/com.sdl.hellosdlandroid I/Sdl Broadcast Receiver: 5.1.1: : Router service app's package: com.sdl.hellosdlandroid
2021-05-28 09:49:20.355 23842-23842/com.sdl.hellosdlandroid I/Sdl Broadcast Receiver: 5.1.1: Not starting device listener, bluetooth device is null and other SDL apps installed.
```

**Comment by joeygrover** [comment]
The issue exists if the app was installed after BT connection or if the app was force closed and not opened until after BT connection. If you believe the patch provided will solve the issue please open a PR, it would be greatly appreciated.

**Comment by ashwink11** [comment]
Hi @joeygrover , I have created PR. could you please review it?

**Comment by jordynmackool** [comment]
Thanks for letting us know that this PR is ready for Livio review. Livio is currently testing for the upcoming 5.2 release on June 30, 2021.

We will then begin reviewing this PR when time allows. Thanks!

**Comment by ashwink11** [comment]
One more way to reproduce this issue.

Environment: 
IVI: Ford SYNC System
Mobile Device: Samsung S8 with Android 9 
Apps: WebEx- latest versions from Play store, SDL Test app with latest SDL library - sdl_android:5+

Steps to reproduce.
1. Connect Ford SYNC and mobile device via Bluetooth.
2. Let SDL apps show on Mobile apps screen on Phone.
3. Restart the mobile device.
4. the Mobile device will automatically connect Bluetooth with SYNC 

Observation: No apps registered on SYNC

**Comment by JulianKast** [comment]
Closing issue as we believe that this issue has already been fixed. When SDLDeviceListener was implemented in 4.12.0, it did require you to cycle Bluetooth if the app was installed after the initial Bluetooth connection was established and therefore did not receive a Bluetooth connection intent while SdlDeviceListener was being used. In 5.2.0 we fixed that issue with this PR: #1685. You may have to have the tdk perform an SDP query by pressing the "Find Mobile Apps" button within a timeout period of 15 seconds of when the app is installed.
